### PR TITLE
AUTOMLALS-158: move `executor` pkg to own pkg in `hyperdrive` repo

### PR DESCRIPTION
**Pull request by johnhouhyper**
Jira ticket: [AUTOMLALS-158](https://gohypergiant.atlassian.net/browse/AUTOMLALS-158)

NOTE: When launching the `cpu-local` container, the `executor` package isn't working because the `hddataclient` (i.e., the data client) hasn't been removed from the `executor` package on this ticket. The removal of the data client is pending PR review on AUTOMLALS-159 ([PR # 21](https://github.com/gohypergiant/mlsdk-image-catalog/pull/21)), so after AUTOMLALS-158 has been reviewed/approved, I'll incorporate the AUTOMLALS-159 changes into `hyperdrive` for review.

### CODE REVIEW

File: docker/Dockerfile.main [hunk 1]
```diff
@@ -11,9 +11,12 @@ FROM ${REGISTRY}/hypergiant-jupyter:cpu-base AS cpu-firefly
 ADD docker/entrypoint-torch.sh /usr/local/bin/
 ADD docker/entrypoint-local.sh /usr/local/bin/
 ADD ./hypertrain /home/jovyan/hypertrain
+ADD ./executor /home/jovyan/executor
+ADD ./executor /tmp/repo
```
Comment (joshuacook) [inline docker/Dockerfile.main:15]: Why add it in two places?

File: executor/executor/__main__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from .walker import Walker
+
+Walker()
```
Comment (joshuacook) [inline executor/executor/__main__.py:3]: Texas Ranger

File: executor/executor/heartbeat/__doc__.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class HeartbeatDoc:
```
Comment (joshuacook) [inline executor/executor/heartbeat/__doc__.py]: Do we still use the Heartbeat? I don't think so

File: executor/executor/walker/.ipynb_checkpoints/__main__-checkpoint.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+from . import Walker
```
Comment (joshuacook) [inline executor/executor/walker/.ipynb_checkpoints/__main__-checkpoint.py:1]: git ignore these ipynb checkpoint files

**Review by joshuacook — APPROVED** [review]
Just a few things then merge

File: docker/Dockerfile.main (hunk 1 above)
Comment (johnhouhyper) [reply docker/Dockerfile.main:15]: Good question.

The `ADD` on line 14 is for `pip install`-ing the `executor` package. After it is installed, we'll delete the `executor` dir (see lines 28-32 of `Dockerfile.main`).

The `ADD` on line 15 is for the `entrypoint-local.sh`, when we copy the demo notebooks, sample data, and yaml files. Please see lines 10-16 of `entrypoint-local.sh`.

File: executor/executor/__main__.py (hunk 2 above)
Comment (johnhouhyper) [reply executor/executor/__main__.py:3]: This just made my day, thank you!

File: executor/executor/heartbeat/__doc__.py (hunk 3 above)
Comment (johnhouhyper) [reply executor/executor/heartbeat/__doc__.py]: Yeah, this is something I noticed also, but wasn't so sure. I didn't see it being used. Thanks for bringing it up, I'll remove it.

**Pull request merged**